backend/models/post.js

```diff
@@ -17,17 +17,24 @@ let postSchema = mongoose.Schema(
       required: true
     },
     author: {
-      type: mongoose.Schema.Types.ObjectId, ref: 'User',
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'User',
       required: true
     },
     categories: {
       type: Array
     },
-    comments: [{
-      type: mongoose.Schema.Types.ObjectId, ref: 'Comment'
-  }],
+    comments: [
+      {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'Comment'
+      }
+    ],
     headerImageUrl: {
       type: String
+    },
+    authorName: {
+      type: String
```

Comment:
I am not sure if this is the best way to go because this is data redundancy. An update anomaly could occur when the user chooses to change/update their name (unlikely to happen in this case, but in projects where data changes a lot, this could be a source of a bug). 

Thereby, you have to update both the post object and the user object. If you do need to update everything.

My recommendation is to create a `controller function` that is named `getAuthorName(authorID)` which will return the result of the query (User matching authorID), and then call that controller function into the route that will _access post_. 

Bundle both the post details and author name
```js
post = { ...post , authorName}
```

Reply:
Fixed. I just used the getUser method instead of creating a separate getAuthorName method.